.github/workflows/turing.yaml

```diff
@@ -102,6 +102,37 @@ jobs:
           path: turing-router.${{ steps.build-image.outputs.router-version }}.tar
           retention-days: ${{ env.ARTIFACT_RETENTION_DAYS }}
 
+  build-test-experiment-engine-plugin:
+    runs-on: ubuntu-latest
+    outputs:
+      test-experiment-engine-plugin-version: ${{ steps.build-image.outputs.test-experiment-engine-plugin-version }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Build Docker image
+        id: build-image
+        working-directory: engines/experiment/examples/plugins/hardcoded
+        env:
+          VERSION: latest
+        run: |
+          set -o pipefail
+          make build-image | tee output.log
+          echo "::set-output name=test-experiment-engine-plugin-version::${{ env.VERSION }}"
+
+      - name: Save Docker image
+        run: |
+          docker image save \
+            --output test-experiment-engine-plugin.${{ steps.build-image.outputs.test-experiment-engine-plugin-version }}.tar \
+            plugin-example-engine-plugin:${{ steps.build-image.outputs.test-experiment-engine-plugin-version }}
```

Comment:
CMIIW, since output `test-experiment-engine-plugin-version` is essentially using `env.VERSION`, is there any reason why we don't set the environment variable `VERSION` at the job level instead, since it's used by all steps except the checkout stage?

Reply:
Yup, this makes sense. I probably just followed the structure of the other `build` jobs, but it's not reasonable for this temporary artifact. Updated